config_app/js/core-config-setup/config-setup-tool.html

```diff
@@ -53,17 +53,25 @@
         </div>
         <div class="co-panel-body">
           <table class="config-table">
+            <tr ng-if="isFieldReadonly('hostname')">
+              <div class="co-alert co-alert-danger">
+                The Hostname configuration is managed externally. These values
+                cannot be changed.
+              </div>
+            </tr>
             <tr>
               <td>Server Hostname:</td>
               <td>
-                <div class="co-alert co-alert-danger" ng-if="config.SERVER_HOSTNAME.indexOf('quay.io') >=0 "
+                <div class="co-alert co-alert-danger" ng-if="config.SERVER_HOSTNAME.indexOf('quay.io') >=0"
                      style="margin-bottom: 10px;">
                   The domain name <code>quay.io</code> is reserved for legacy reasons and cannot be used in
                   <span class="registry-name"></span> installations.
                 </div>
-
-                <span class="config-string-field" binding="config.SERVER_HOSTNAME"
-                      placeholder="Hostname (and optional port if non-standard)" pattern="{{ HOSTNAME_REGEX }}"></span>
+                <span class="config-string-field"
+                      binding="config.SERVER_HOSTNAME"
+                      placeholder="Hostname (and optional port if non-standard)"
+                      pattern="{{ HOSTNAME_REGEX }}"
+                      is-readonly="isFieldReadonly('hostname')"></span>
```

Comment:
Use [one-time binding](https://docs.angularjs.org/guide/expression#one-time-binding) to improve performance (`::isFieldReadonly(...)`).